src/app/Fake.Runtime/SdkAssemblyResolver.fs

```diff
@@ -143,23 +143,28 @@ type SdkAssemblyResolver(logLevel: Trace.VerboseLevel) =
         if this.LogLevel.PrintVerbose then
             Trace.tracefn "Trying to resolve runtime version from network.."
 
+        let sdkSet = this.SdkVersionRaws |> Set.ofList
+
         try
-            let sdkVersionReleases =
-                ProductCollection.GetAsync()
-                |> Async.AwaitTask
-                |> Async.RunSynchronously
-                |> List.ofSeq
-                |> List.filter (fun product ->
-                    this.SdkVersionRaws
-                    |> List.exists (fun raws -> product.ProductVersion.Equals raws))
-
-            sdkVersionReleases
-            |> List.collect (fun rel ->
-                rel.GetReleasesAsync()
-                |> Async.AwaitTask
-                |> Async.RunSynchronously
-                |> List.ofSeq)
+            task {
+                let! productCollection = ProductCollection.GetAsync()
+
+                let sdkVersionReleases =
+                    productCollection
+                    |> Seq.filter (fun product -> sdkSet |> Set.exists (fun raws -> product.ProductVersion.Equals raws))
+
+                let! releases =
+                    sdkVersionReleases
+                    |> Seq.map (fun rel -> rel.GetReleasesAsync())
+                    |> System.Threading.Tasks.Task.WhenAll
+
+                return releases |> Seq.collect (fun sdks -> sdks |> Seq.toList)
```

Comment:
Maybe without lists allocation?
```suggestion
                return releases |> Seq.collect id
```

Reply:
The issue it's a "list of arrays" so either Seq.toArray |> Array.collect |> Array.toList or this. As a list is small, like max 10, I don't see a difference.

Reply:
Will not it implicitly convert arrays to seqs?

Reply:
it didn't